esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -0,0 +1,202 @@
+    """General CORDEX grid fix."""
+
+    _grid_diff_msg = ('Maximum difference between original {} '
+                      'points and standard {} domain points '
+                      'for dataset {} and driver {} is: {}')
+
+    def _check_grid_differences(self, old_coord, new_coord):
+        """Check differences between coords."""
+        diff = np.max(np.abs(old_coord.points - new_coord.points))
+        logger.debug(
+            self._grid_diff_msg,
+            new_coord.var_name,
+            self.extra_facets['domain'],
+            self.extra_facets['dataset'],
+            self.extra_facets['driver'],
+            diff
+        )
+
+        if diff > 10e-4:
+            raise RecipeError(
+                    "Differences between the original grid and the "
+                    f"standarised grid are above 10e-4 {new_coord.units}.",
+                )
+
+    def _fix_rotated_coords(self, cube):
+        """Fix rotated coordinates."""
+        data_domain = self.extra_facets['domain']
+        domain = cx.cordex_domain(data_domain, add_vertices=True)
+        domain_info = cx.domain_info(data_domain)
+        for dim_coord in ['rlat', 'rlon']:
+            old_coord = cube.coord(domain[dim_coord].standard_name)
+            old_coord_dims = old_coord.cube_dims(cube)
+            points = domain[dim_coord].data
+            coord_system = iris.coord_systems.RotatedGeogCS(
+                grid_north_pole_latitude=domain_info['pollat'],
+                grid_north_pole_longitude=domain_info['pollon']
+            )
+            new_coord = iris.coords.DimCoord(
+                points,
+                var_name=dim_coord,
+                standard_name=domain[dim_coord].standard_name,
+                long_name=domain[dim_coord].long_name,
+                units=Unit('degrees'),
+                coord_system=coord_system,
+            )
+            self._check_grid_differences(old_coord, new_coord)
+            new_coord.guess_bounds()
+            cube.remove_coord(old_coord)
+            cube.add_dim_coord(new_coord, old_coord_dims)
+
+    def _fix_geographical_coords(self, cube):
+        """Fix geographical coordinates."""
+        data_domain = self.extra_facets['domain']
+        domain = cx.cordex_domain(data_domain, add_vertices=True)
+        for aux_coord in ['lat', 'lon']:
+            old_coord = cube.coord(domain[aux_coord].standard_name)
+            cube.remove_coord(old_coord)
+            points = domain[aux_coord].data
+            bounds = domain[f'{aux_coord}_vertices'].data
+            new_coord = iris.coords.AuxCoord(
+                points,
+                var_name=aux_coord,
+                standard_name=domain[aux_coord].standard_name,
+                long_name=domain[aux_coord].long_name,
+                units=Unit(domain[aux_coord].units),
+                bounds=bounds
+            )
+            logger.info(old_coord.units)
+            self._check_grid_differences(old_coord, new_coord)
```

Comment:
A value conversion for longitudes should be added before this check, because if the longitudes are given in different ranges, the check will report an error when it may not be the case

Reply:
Actually this case only happens for one dataset and driver and I think the check is right to be raised, because the longitude values are quite wrong.